src/main/resources/io/jenkins/plugins/oidc_provider/config/IdTokenConfiguration/config.jelly

```diff
@@ -28,31 +28,31 @@ THE SOFTWARE.
     <f:section title="OpenID Connect">
         <f:advanced title="${%Claim templates}" align="left">
             <f:entry field="claimTemplates" title="${%General claim templates}">
-                <f:repeatableProperty field="claimTemplates">
+                <f:repeatableHeteroProperty field="claimTemplates" honorOrder="true" hasHeader="true">
                     <f:block>
                         <div align="right">
                             <f:repeatableDeleteButton/>
                         </div>
                     </f:block>
-                </f:repeatableProperty>
+                </f:repeatableHeteroProperty>
             </f:entry>
             <f:entry field="buildClaimTemplates" title="${%Build-scoped claim templates}">
```

Comment:
Part of my proposal was to actually remove `buildClaimTemplates` and `globalClaimTemplates`, and change the default claim templates list to use the new required variable features instead. Could probably retain a measure of compatibility in case there is anyone already using the new system in production, though I doubt it.

Reply:
I have clients who use this in production currently, so I dont think we should make any breaking changes as part of this PR.

Reply:
Should still be possible to deprecate the old fields and remove them from GUI config, presuming that there is no remaining legitimate use case. Certainly `buildClaimTemplates` can simply be redefined as `claimTemplates` with, e.g., `BUILD_ID` required. `globalClaimTemplates` is trickier since this is a negative test, and the current trunk behavior presumes that there are no overlaps between claims.

Reply:
Do you just want 
```java
@Deprecated
private @CheckForNull List<ClaimTemplate> buildClaimTemplates;
@Deprecated
private @CheckForNull List<ClaimTemplate> globalClaimTemplates;
```

And the components removed from the config.jelly?

Reply:
I don't know how to remove these without introducing breaking changes to jcasc etc

Reply:
Not sure offhand, I would need to study it. No time at the moment I am afraid.

Reply:
Does the deprecation have to be part of this PR? I'd like to see this move forward (as long as its good enough ofc)